pkg/v1/platform.go

```diff
@@ -85,6 +85,47 @@ func (p Platform) Equals(o Platform) bool {
 		stringSliceEqualIgnoreOrder(p.Features, o.Features)
 }
 
+// Satisfies returns true if this Platform "satisfies" the given spec Platform.
+//
+// Note that this is different from Equals and that Satisfies is not reflexive.
+//
+// The given spec represents "requirements" such that any missing values in the
+// spec are not compared.
+//
+// For OSFeatures and Features, Satisfies will return true if this Platform's
+// fields contain a superset of the values in the spec's fields (order ignored).
+func (p Platform) Satisfies(spec Platform) bool {
+	return satisfies(spec.OS, p.OS) &&
+		satisfies(spec.Architecture, p.Architecture) &&
+		satisfies(spec.Variant, p.Variant) &&
+		satisfies(spec.OSVersion, p.OSVersion) &&
+		satisfiesList(spec.OSFeatures, p.OSFeatures) &&
+		satisfiesList(spec.Features, p.Features)
```

Comment:
Can we just leave features and osfeatures in the dustbin of history? Are they ever practically used?

Reply:
I wanna say no just because OSFeatures is in the spec. We could probably drop Features, but it's no work to keep it given that we have to implement OSFeatures anyway.

Reply:
It sounds like Jason's volunteering to open a spicy image-spec discussion :eyes: :hot_pepper:

(`features` is in the spec too: https://github.com/opencontainers/image-spec/blob/24027e1093941c9785f19475c8c1a4a8bef23c29/image-index.md?plain=1#L48-L86)